genshi/filters/i18n.py

```diff
@@ -711,8 +711,9 @@ def __call__(self, stream, ctxt=None, translate_text=True,
                 for name, value in attrs:
                     newval = value
                     if isinstance(value, six.string_types):
-                        if translate_attrs and name in include_attrs:
-                            newval = gettext(value)
+                        text = value.strip()
```

Comment:
It took me awhile to understand why `value.strip()` is needed (because the value is stripped by Genshi during extraction) and why `and text` is needed (because `gettext` doesn't support being passed empty strings), so thank you very much to everyone who supplied explanations in the comments.